src/lightgbm/src/main/scala/LightGBMRegressor.scala

```diff
@@ -85,10 +85,13 @@ class LightGBMRegressor(override val uid: String)
       getVerbosity, categoricalIndexes, getBoostFromAverage, getBoostingType)
     log.info(s"LightGBMRegressor parameters: ${trainParams.toString}")
     val networkParams = NetworkParams(getDefaultListenPort, inetAddress, port)
-
+    val validationData =
+      if (get(validationIndicatorCol).isDefined && get(validationIndicatorCol).get.nonEmpty)
```

Comment:
will this ever be an empty string? I think if you use an empty string we should remove that API and just go with setting or not setting and then turn this into a map on an option

Reply:
it's just the column name. Supposedly it should never be an empty string but if the user for some bizarre reason sets it to empty string we can just ignore it.  Actually, this is a bad check.  I should instead check if the column exists on the dataset.  I modified the check.